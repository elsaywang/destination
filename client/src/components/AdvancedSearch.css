:local(.reportSuitesFilter) {
    display: flex;
    align-items: center;
    margin-bottom: 10px;
}
