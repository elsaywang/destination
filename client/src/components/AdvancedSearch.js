import React from 'react';
import ComboBox from '@react/react-spectrum/ComboBox';
import Switch from '@react/react-spectrum/Switch';

function AdvancedSearch({ enabled, onAdvancedSearchChange, onFilterChange }) {
    const filterOptions = [];

    return (
        <div>
            <Switch
                onChange={onAdvancedSearchChange}
                checked={enabled}
                aria-label="Advanced Search"
                label="Advanced search for Adobe Analytics. Search by key/value names and results in:"
            />
            <ComboBox
                placeholder="Filter by report suite"
                onChange={onFilterChange}
                options={filterOptions}
                disabled={!enabled}
                quiet
            />
        </div>
    );
}

export default AdvancedSearch;
