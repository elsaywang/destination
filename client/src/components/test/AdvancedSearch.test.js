import React from 'react';
import { shallow } from 'enzyme';
import AdvancedSearch from '../AdvancedSearch';
import ComboBox from '@react/react-spectrum/ComboBox';

describe('Advanced Search', () => {
    const mockFn = jest.fn();
    const reportSuites = [
        {
            dataSourceId: 168815,
            pid: 1194,
            name: 'Test Report Suite Edited1505153440289',
            suite: 'test-report-suite-edited1505153440289',
            datacenter: 'sin',
        },
        {
            dataSourceId: 169626,
            pid: 1194,
            name: 'Test Report Suite Edited1505154745046',
            suite: 'test-report-suite-edited1505154745046',
            datacenter: 'sin',
        },
    ];
    const wrapper = shallow(
        <AdvancedSearch
            sourceName="test"
            reportSuites={reportSuites}
            enabled={false}
            onReportSuiteChange={mockFn}
            onReportSuiteSelect={mockFn}
            onAdvancedSearchChange={mockFn}
        />,
    );

    describe('rendering', () => {
        it('matches snapshot', () => {
            expect(wrapper).toMatchSnapshot();
        });

        it('renders <ComboBox />', () => {
            expect(wrapper.find(ComboBox).exists()).toBe(true);
        });

        it('renders <ComboBox /> with disabled: true when <AdvancedSearch /> is passed enabled: false', () => {
            wrapper.setProps({ enabled: false });
            expect(wrapper.find(ComboBox).props().disabled).toBe(true);
        });

        it('renders <ComboBox /> with disabled: false when <AdvancedSearch /> is passed enabled: true', () => {
            wrapper.setProps({ enabled: true });
            expect(wrapper.find(ComboBox).props().disabled).toBe(false);
        });

        it('renders <ComboBox /> with value passed in', () => {
            wrapper.setProps({ sourceName: 'abc' });
            expect(wrapper.find(ComboBox).props().value).toBe('abc');
        });
    });
});
