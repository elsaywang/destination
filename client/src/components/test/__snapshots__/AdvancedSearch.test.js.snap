// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Advanced Search rendering matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AdvancedSearch
    enabled={false}
    onAdvancedSearchChange={[MockFunction]}
    onFilterChange={[MockFunction]}
    onFilterSelect={[MockFunction]}
    reportSuites={
      Array [
        Object {
          "dataSourceId": 168815,
          "datacenter": "sin",
          "name": "Test Report Suite Edited1505153440289",
          "pid": 1194,
          "suite": "test-report-suite-edited1505153440289",
        },
        Object {
          "dataSourceId": 169626,
          "datacenter": "sin",
          "name": "Test Report Suite Edited1505154745046",
          "pid": 1194,
          "suite": "test-report-suite-edited1505154745046",
        },
      ]
    }
    sourceName="test"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span>
          Search and see results in friendly names:  
        </span>,
        <ComboBox
          data-test="advanced-search-filter"
          disabled={true}
          invalid={false}
          onChange={[MockFunction]}
          onSelect={[Function]}
          options={
            Array [
              "Test Report Suite Edited1505153440289",
              "Test Report Suite Edited1505154745046",
            ]
          }
          placeholder="Filter by report suite"
          quiet={true}
          required={false}
          value=""
        />,
      ],
      "className": undefined,
      "data-test": "advanced-search",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Search and see results in friendly names:  ",
        },
        "ref": null,
        "rendered": "Search and see results in friendly names:  ",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "data-test": "advanced-search-filter",
          "disabled": true,
          "invalid": false,
          "onChange": [MockFunction],
          "onSelect": [Function],
          "options": Array [
            "Test Report Suite Edited1505153440289",
            "Test Report Suite Edited1505154745046",
          ],
          "placeholder": "Filter by report suite",
          "quiet": true,
          "required": false,
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span>
            Search and see results in friendly names:  
          </span>,
          <ComboBox
            data-test="advanced-search-filter"
            disabled={true}
            invalid={false}
            onChange={[MockFunction]}
            onSelect={[Function]}
            options={
              Array [
                "Test Report Suite Edited1505153440289",
                "Test Report Suite Edited1505154745046",
              ]
            }
            placeholder="Filter by report suite"
            quiet={true}
            required={false}
            value=""
          />,
        ],
        "className": undefined,
        "data-test": "advanced-search",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Search and see results in friendly names:  ",
          },
          "ref": null,
          "rendered": "Search and see results in friendly names:  ",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "data-test": "advanced-search-filter",
            "disabled": true,
            "invalid": false,
            "onChange": [MockFunction],
            "onSelect": [Function],
            "options": Array [
              "Test Report Suite Edited1505153440289",
              "Test Report Suite Edited1505154745046",
            ],
            "placeholder": "Filter by report suite",
            "quiet": true,
            "required": false,
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
